### DISCUSSION TITLE
Seed Image with Project Partially Installed

### DISCUSSION THREAD

**Discussion by seandenigris** [hpi-swa/smalltalkCI#658] [Q&A]
I have a few projects that are largely consist of one primary dependent project. To save CI time (which is limited for private GH projects), I was thinking of using the build artifact from the dependency as the base image in which to complete the installation. I noticed SmalltalkCI has an `-i` flag to supply an image, which seems like what I need.

Is the plan above feasible? If so, is there an example of how to go about it? If not, any pointers?

**Comment by fniephaus** [comment]
Hi @seandenigris, sorry for the super slow response and no, I haven't seen anyone do this before. Code installation from disk is typically very fast, so using a prepared image only seems to make sense if you can save a lot of work. Do you have such a use case? Can you share some rough numbers on how longer the different stages take in your CI system?

Alternatively, you could set up a self-hosted runner and use that for your private repos.

**Reply by seandenigris** [reply]
Thanks for getting back to me! 

As an example, a complex project that is the base for many others is my Dynabook project. CI runs seem to take 5-14 minutes ([see here](https://github.com/seandenigris/Dynabook/actions)). Since I do a lot of small atomic commits, this definitely adds up.

Conceptually, would the `-i` flag allow me to do this or am I misunderstanding what that flag does?

**Reply by fniephaus** [reply]
First of all, it seems the option is called `--image` and if you use it, smalltalkCI will just use it instead of the default image. So you could give this a try, pre-install packages that barely change so that Metacello only needs to load and install a couple of dependencies.

It seems you are installing quite a few external dependencies, so what you could also try is [cache the dependencies on disk](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#using-the-cache-action). Instead of Metacello cloning a bunch of repos, the cache would just download everything in one go. The question is, what to cache, but that is probably something you can answer.

Hope this helps!
